src/Select/Select.test.js

```diff
@@ -85,33 +85,123 @@ describe('<Select />', () => {
     });
 
     describe('interactions', () => {
-        let onSelect;
-        let element;
-        beforeEach(() => {
-            onSelect = jest.fn();
-            element = mount(
-                <Select onSelect={onSelect} options={options} />
+        describe('onSelect', () => {
+            let onSelect;
+            let element;
+            beforeEach(() => {
+                onSelect = jest.fn();
+                element = mount(
+                    <Select onSelect={onSelect} options={options} />
+                );
+                element.find('.fd-select__button').simulate('click');
+            });
+
+            afterEach(() => {
+                let event = new MouseEvent('mousedown', {});
+                document.dispatchEvent(event);
+            });
+
+            test('should call onSelect when the first checkbox option is clicked', () => {
+                element.find('.fd-list__item').at(0).simulate('click');
+
+                expect(onSelect).toHaveBeenCalledTimes(1);
+                expect(onSelect).toHaveBeenCalledWith(expect.anything(), options[0]);
+            });
+
+            test('should call onSelect when the second checkbox option is clicked', () => {
+                element.find('.fd-list__item').at(1).simulate('click');
+
+                expect(onSelect).toHaveBeenCalledTimes(1);
+                expect(onSelect).toHaveBeenCalledWith(expect.anything(), options[1]);
+            });
+        });
+
+        test('should trigger callback, with valid event object as argument, on input onFocus event', async() => {
+            const focusSpy = jest.fn();
+            const wrapper = setup({
+                onFocus: focusSpy
+            });
+
+            await act(async() => {
+                wrapper.find('.fd-select').simulate('focus');
+            });
+
+            expect(focusSpy).toHaveBeenCalledTimes(1);
+
+            expect(
+                focusSpy
+            ).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    target: wrapper.find('.fd-select').getDOMNode()
+                })
```

Comment:
These onSelect tests remain unchanged - I put them inside their own describe block so we could keep all the interactions in the interaction describe block